docs/testing_mbed_OS_5.md

```diff
@@ -0,0 +1,187 @@
+# Testing in mbed OS 5
+
+The way tests are ran and compiled in mbed OS 5 is substantially different than in previous versions of mbed. Previously tests were located in one known location and a python file was kept (`tools/tests.py`) keeping track of their dependencies, capabilities, and configurations. mbed OS 5 has adopted a more distributed approach to testing. Now test code lives alongside the test code, and these are dynamically discovered by the test tools.
+
+## Table of Contents
+
+- [Using tests](#using-tests)
+  - [Test code structure](#test-code-structure)
+    - [Test discovery](#test-discovery)
+    - [Test names](#test-names)
+  - [Building tests](#building-tests)
+  - [Running tests](#running-tests)
+  - [Writing tests](#writing-tests)
+- [Debugging tests](#debugging-tests)
+- [Known issues](#known-issues)
+
+## Test code structure
+
+Tests can exist throughout mbed OS and your project's code. They are located under a special directory called `TESTS` (case is important!).
+
+Placing code under this directory means it will be ignored when building applications and libraries. This code is only ever used when building tests. This is important since all tests require a `main()` function, and building it with your application would cause multiple `main()` functions to be defined.
+
+In addition to being placed under a `TESTS` directory, test sources must exist under two other directories: a "test group" folder and a "test case" folder. The following is an example of this structure:
```

Comment:
`TESTS` refers to an actual directory in the code base. but "test group" and "test case" are just names for the different levels of the folders. They actual name of the directories can be anything.

It sounds like this might not be 100% clear. Maybe I should just remove the `"` ?